src/typography/src/Heading.tsx

```diff
@@ -1,32 +1,40 @@
-import React, { PureComponent } from 'react'
+import React, { PureComponent, Validator } from 'react'
 import PropTypes from 'prop-types'
 import Box from 'ui-box'
-import { withTheme } from '../../theme'
+import { withTheme, Theme } from '../../theme'
 
-class Heading extends PureComponent<any & React.ComponentProps<typeof Box>> {
-  static propTypes = {
-    /**
-     * The size of the heading.
-     */
-    size: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700, 800, 900])
-      .isRequired,
+type HeadingSize = 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900
+
+export interface HeadingProps extends React.ComponentProps<typeof Box> {
+  /**
+   * Pass `default` to use the default margin top for that size.
+   */
+  marginTop?: boolean | number | string | 'default'
 
-    /**
-     * Pass `default` to use the default margin top for that size.
-     */
+  /**
+   * The size of the heading.
+   */
+  size: HeadingSize
+
+  /**
+   * Theme provided by ThemeProvider.
+   */
+  theme: Theme
+}
+
+class Heading extends PureComponent<HeadingProps> {
+  static propTypes = {
     marginTop: PropTypes.oneOfType([
       PropTypes.bool,
       PropTypes.number,
       PropTypes.string
     ]),
-
-    /**
-     * Theme provided by ThemeProvider.
-     */
-    theme: PropTypes.object.isRequired
+    size: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700, 800, 900])
```

Comment:
I wonder if instead of `size: PropTypes.oneOf([100, 200, 300, ...])` we could have `size: PropTypes.oneOf([size in HeadingSize]: number)`. Idea from [here](https://www.typescriptlang.org/docs/handbook/advanced-types.html) (I'm sorry, I couldn't find a way to link directly to it).

Reply:
Is it possible to enumerate the types as a value? I couldn't find the reference in the doc you linked.

Reply:
Under `Mapped types` I saw this:

```
type Keys = 'option1' | 'option2';
type Flags = { [K in Keys]: boolean };
```

I'm not sure if this is the same case, but it looked similar to me.

Reply:
Right, but those are TypeScript types. I don't think you can use types as values in your code.

Reply:
That makes sense. Thank you! I'm sorry for the back and forth. :)